apps/makescene/makescene.cc

```diff
@@ -423,12 +426,55 @@ import_bundle_nvm_or_colmap (AppSettings const& conf, bool load_nvm = true)
         int const maxdim = std::max(image->width(), image->height());
         mve_cam.flen = mve_cam.flen / static_cast<float>(maxdim);
 
-        mve::ByteImage::Ptr undist = mve::image::image_undistort_vsfm<uint8_t>
-            (std::dynamic_pointer_cast<mve::ByteImage const>(image), 
-            mve_cam.flen, 
-            cam_info.radial_distortion);
+        mve::ByteImage::Ptr undist;
+        if (load_nvm)
+        {
+            undist = mve::image::image_undistort_vsfm<uint8_t>
+                (std::dynamic_pointer_cast<mve::ByteImage const>(image),
+                mve_cam.flen,
+                cam_info.radial_distortion);
+        }
+        else
+        {
+            // For Colmap reconstructions, MVE expects camera models without
+            // radial distortion. Potentially, obtained in Colmap's undistortion
+            // step.
+            assert(cam_info.radial_distortion == 0.0);
```

Comment:
I'm not sure if asserting provides appropriate communication to the user. Should this instead be an if, a message, and an std::exit(1) or something?

Reply:
This assertion is really considered as assertion. It should be always true, since other cases are detected earlier in [ libs/mve/bundle_io.cc at line 565](https://github.com/simonfuhrmann/mve/pull/514/commits/eec565e886899815eb88e294e8e19374b77f463b#diff-b48aa0fdef49e807586f79bef370769dL565). 
     
```
        std::string msg = "Unsupported camera model with radial distortion "
            "detected! If possible, re-run the SfM reconstruction with the "
            "SIMPLE_PINHOLE or the PINHOLE camera model (recommended). " 
            "Otherwise, use the undistortion step in Colmap to obtain "
            "undistorted images and corresponding camera models without radial "
            "distortion.";

        throw util::Exception(msg.c_str());
```

Is throwing an exception there okay? Or should it be replaced with `std::exit(1)`?
Side note: Other functions in `bundle_io.cc` use exceptions for similar cases (and not `std::exit(1)`)

Reply:
If this is really just an assertion, this is fine then.

Reply:
On another note, exceptions are a thing for libraries, and we don't want to call std::exit() in a library. This is a CLI, and hence responsible for catching all exceptions and making sure a nice message is printed to the user.